[2022-10-28T19:11:12.428+0000] {taskinstance.py:1165} INFO - Dependencies all met for <TaskInstance: ENG_hourly_process_v1.calculate_hourly_demand_forecast scheduled__2022-10-26T16:05:00+00:00 [queued]>
[2022-10-28T19:11:12.681+0000] {taskinstance.py:1165} INFO - Dependencies all met for <TaskInstance: ENG_hourly_process_v1.calculate_hourly_demand_forecast scheduled__2022-10-26T16:05:00+00:00 [queued]>
[2022-10-28T19:11:12.681+0000] {taskinstance.py:1362} INFO - 
--------------------------------------------------------------------------------
[2022-10-28T19:11:12.681+0000] {taskinstance.py:1363} INFO - Starting attempt 1 of 2
[2022-10-28T19:11:12.696+0000] {taskinstance.py:1364} INFO - 
--------------------------------------------------------------------------------
[2022-10-28T19:11:14.312+0000] {taskinstance.py:1383} INFO - Executing <Task(PythonOperator): calculate_hourly_demand_forecast> on 2022-10-26 16:05:00+00:00
[2022-10-28T19:11:14.577+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'ENG_hourly_process_v1', 'calculate_hourly_demand_forecast', 'scheduled__2022-10-26T16:05:00+00:00', '--job-id', '96', '--raw', '--subdir', 'DAGS_FOLDER/ENG_hourly_process_dag.py', '--cfg-path', '/tmp/tmpchtw55ip']
[2022-10-28T19:11:14.609+0000] {standard_task_runner.py:83} INFO - Job 96: Subtask calculate_hourly_demand_forecast
[2022-10-28T19:11:14.528+0000] {standard_task_runner.py:55} INFO - Started process 6273 to run task
[2022-10-28T19:11:23.200+0000] {task_command.py:376} INFO - Running <TaskInstance: ENG_hourly_process_v1.calculate_hourly_demand_forecast scheduled__2022-10-26T16:05:00+00:00 [running]> on host f011703e6eb5
[2022-10-28T19:11:24.434+0000] {taskinstance.py:1592} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=colmo786@gmail.com
AIRFLOW_CTX_DAG_OWNER=energy
AIRFLOW_CTX_DAG_ID=ENG_hourly_process_v1
AIRFLOW_CTX_TASK_ID=calculate_hourly_demand_forecast
AIRFLOW_CTX_EXECUTION_DATE=2022-10-26T16:05:00+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2022-10-26T16:05:00+00:00
[2022-10-28T19:11:24.979+0000] {logging_mixin.py:120} WARNING - /home/***/.local/lib/python3.7/site-packages/***/utils/context.py:297 AirflowContextDeprecationWarning: Accessing 'execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_start' or 'logical_date' instead.
[2022-10-28T19:11:24.979+0000] {logging_mixin.py:120} INFO - INFO calculate_hourly_demand_forecast execution init:  2022-10-26T16:05:00+00:00
[2022-10-28T19:11:24.980+0000] {logging_mixin.py:120} INFO - WORKDIR: /opt/***
[2022-10-28T19:11:25.141+0000] {logging_mixin.py:120} INFO - INFO Module build_postgres_cnxn: DB Connection to host local_pgdb Ok
[2022-10-28T19:11:25.298+0000] {logging_mixin.py:120} INFO - INFO pg_select_to_pandas: query executed Ok. Number of records returned: 1
[2022-10-28T19:11:25.683+0000] {logging_mixin.py:120} INFO - INFO pg_select_to_pandas: query executed Ok. Number of records returned: 49
[2022-10-28T19:12:14.834+0000] {logging_mixin.py:120} INFO - 1/1 [==============================] - ETA: 0s
[2022-10-28T19:12:14.835+0000] {logging_mixin.py:120} INFO - 1/1 [==============================] - 11s 11s/step
[2022-10-28T19:12:14.869+0000] {logging_mixin.py:120} INFO - INFO _calculate_hourly_demand_forecast: forecast calculated for next 24 hours. Since: 2022-10-28 15:00
[2022-10-28T19:12:26.493+0000] {logging_mixin.py:120} INFO - INFO _calculate_hourly_demand_forecast - 24 records were upserted. Table cammesa_db.hourly_demand_forecast.
[2022-10-28T19:12:26.494+0000] {python.py:177} INFO - Done. Returned value was: None
[2022-10-28T19:12:26.565+0000] {taskinstance.py:1406} INFO - Marking task as SUCCESS. dag_id=ENG_hourly_process_v1, task_id=calculate_hourly_demand_forecast, execution_date=20221026T160500, start_date=20221028T191112, end_date=20221028T191226
[2022-10-28T19:12:27.117+0000] {local_task_job.py:164} INFO - Task exited with return code 0
[2022-10-28T19:12:27.353+0000] {local_task_job.py:273} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2022-10-28T22:39:37.827+0000] {taskinstance.py:1165} INFO - Dependencies all met for <TaskInstance: ENG_hourly_process_v1.calculate_hourly_demand_forecast scheduled__2022-10-26T16:05:00+00:00 [queued]>
[2022-10-28T22:39:37.946+0000] {taskinstance.py:1165} INFO - Dependencies all met for <TaskInstance: ENG_hourly_process_v1.calculate_hourly_demand_forecast scheduled__2022-10-26T16:05:00+00:00 [queued]>
[2022-10-28T22:39:37.946+0000] {taskinstance.py:1362} INFO - 
--------------------------------------------------------------------------------
[2022-10-28T22:39:37.946+0000] {taskinstance.py:1363} INFO - Starting attempt 1 of 2
[2022-10-28T22:39:37.947+0000] {taskinstance.py:1364} INFO - 
--------------------------------------------------------------------------------
[2022-10-28T22:39:39.858+0000] {taskinstance.py:1383} INFO - Executing <Task(PythonOperator): calculate_hourly_demand_forecast> on 2022-10-26 16:05:00+00:00
[2022-10-28T22:39:39.980+0000] {standard_task_runner.py:55} INFO - Started process 1161 to run task
[2022-10-28T22:39:40.077+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'ENG_hourly_process_v1', 'calculate_hourly_demand_forecast', 'scheduled__2022-10-26T16:05:00+00:00', '--job-id', '20', '--raw', '--subdir', 'DAGS_FOLDER/ENG_hourly_process_dag.py', '--cfg-path', '/tmp/tmp8sj5ukh8']
[2022-10-28T22:39:40.080+0000] {standard_task_runner.py:83} INFO - Job 20: Subtask calculate_hourly_demand_forecast
[2022-10-28T22:40:23.873+0000] {task_command.py:376} INFO - Running <TaskInstance: ENG_hourly_process_v1.calculate_hourly_demand_forecast scheduled__2022-10-26T16:05:00+00:00 [running]> on host 74ce152bc4b9
[2022-10-28T22:40:25.952+0000] {taskinstance.py:1592} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=colmo786@gmail.com
AIRFLOW_CTX_DAG_OWNER=energy
AIRFLOW_CTX_DAG_ID=ENG_hourly_process_v1
AIRFLOW_CTX_TASK_ID=calculate_hourly_demand_forecast
AIRFLOW_CTX_EXECUTION_DATE=2022-10-26T16:05:00+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2022-10-26T16:05:00+00:00
[2022-10-28T22:40:26.092+0000] {logging_mixin.py:120} WARNING - /home/***/.local/lib/python3.7/site-packages/***/utils/context.py:297 AirflowContextDeprecationWarning: Accessing 'execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_start' or 'logical_date' instead.
[2022-10-28T22:40:26.092+0000] {logging_mixin.py:120} INFO - INFO calculate_hourly_demand_forecast execution init:  2022-10-26T16:05:00+00:00
[2022-10-28T22:40:26.093+0000] {logging_mixin.py:120} INFO - WORKDIR: /opt/***
[2022-10-28T22:40:26.233+0000] {logging_mixin.py:120} INFO - INFO Module build_postgres_cnxn: DB Connection to host local_pgdb Ok
[2022-10-28T22:40:26.361+0000] {local_task_job.py:87} ERROR - Received SIGTERM. Terminating subprocesses
[2022-10-28T22:40:26.435+0000] {process_utils.py:133} INFO - Sending Signals.SIGTERM to group 1161. PIDs of all processes in the group: [1161]
[2022-10-28T22:40:26.436+0000] {process_utils.py:84} INFO - Sending the signal Signals.SIGTERM to group 1161
[2022-10-28T22:40:26.537+0000] {taskinstance.py:1562} ERROR - Received SIGTERM. Terminating subprocesses.
[2022-10-28T22:40:26.608+0000] {logging_mixin.py:120} INFO - ERROR pg_select_to_pandas: Error executing query on host: local_pgdb database user query: SELECT MAX(timestamp) AS max_timestamp
                    FROM cammesa_db.hourly_demand 
                
Traceback (most recent call last):   File "/opt/***/dags/energy.py", line 87, in pg_select_to_pandas     df = pd.DataFrame(data=data, columns=colnames)   File "/home/***/.local/lib/python3.7/site-packages/pandas/core/frame.py", line 708, in __init__     typ=manager,   File "/home/***/.local/lib/python3.7/site-packages/pandas/core/internals/construction.py", line 136, in arrays_to_mgr     arrays, arr_names, axes, consolidate=consolidate   File "/home/***/.local/lib/python3.7/site-packages/pandas/core/internals/managers.py", line 1774, in create_block_manager_from_arrays     mgr = BlockManager(blocks, axes)   File "/home/***/.local/lib/python3.7/site-packages/pandas/core/internals/managers.py", line 914, in __init__     self._verify_integrity()   File "/home/***/.local/lib/python3.7/site-packages/pandas/core/internals/managers.py", line 918, in _verify_integrity     tot_items = sum(len(x.mgr_locs) for x in self.blocks)   File "/home/***/.local/lib/python3.7/site-packages/pandas/core/internals/managers.py", line 918, in <genexpr>     tot_items = sum(len(x.mgr_locs) for x in self.blocks)   File "/home/***/.local/lib/python3.7/site-packages/pandas/core/internals/blocks.py", line 238, in mgr_locs     @property   File "/home/***/.local/lib/python3.7/site-packages/***/models/taskinstance.py", line 1564, in signal_handler     raise AirflowException("Task received SIGTERM signal") ***.exceptions.AirflowException: Task received SIGTERM signal
[2022-10-28T22:40:26.609+0000] {logging_mixin.py:120} INFO - ERROR nombre_del_modulo calling subprocess: ERROR pg_select_to_pandas: Error executing query on host: local_pgdb database user query: SELECT MAX(timestamp) AS max_timestamp
                    FROM cammesa_db.hourly_demand 
                
Traceback (most recent call last):   File "/opt/***/dags/energy.py", line 87, in pg_select_to_pandas     df = pd.DataFrame(data=data, columns=colnames)   File "/home/***/.local/lib/python3.7/site-packages/pandas/core/frame.py", line 708, in __init__     typ=manager,   File "/home/***/.local/lib/python3.7/site-packages/pandas/core/internals/construction.py", line 136, in arrays_to_mgr     arrays, arr_names, axes, consolidate=consolidate   File "/home/***/.local/lib/python3.7/site-packages/pandas/core/internals/managers.py", line 1774, in create_block_manager_from_arrays     mgr = BlockManager(blocks, axes)   File "/home/***/.local/lib/python3.7/site-packages/pandas/core/internals/managers.py", line 914, in __init__     self._verify_integrity()   File "/home/***/.local/lib/python3.7/site-packages/pandas/core/internals/managers.py", line 918, in _verify_integrity     tot_items = sum(len(x.mgr_locs) for x in self.blocks)   File "/home/***/.local/lib/python3.7/site-packages/pandas/core/internals/managers.py", line 918, in <genexpr>     tot_items = sum(len(x.mgr_locs) for x in self.blocks)   File "/home/***/.local/lib/python3.7/site-packages/pandas/core/internals/blocks.py", line 238, in mgr_locs     @property   File "/home/***/.local/lib/python3.7/site-packages/***/models/taskinstance.py", line 1564, in signal_handler     raise AirflowException("Task received SIGTERM signal") ***.exceptions.AirflowException: Task received SIGTERM signal
[2022-10-28T22:40:26.616+0000] {taskinstance.py:1851} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/opt/airflow/dags/energy.py", line 84, in pg_select_to_pandas
    cursor.execute(sql_query)
psycopg2.InterfaceError: cursor already closed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 175, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 193, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/ENG_hourly_process_dag.py", line 63, in calculate_hourly_demand_forecast
    , port=port, n_lookback=48, n_forecast=24, verbose=True)
  File "/opt/airflow/dags/energy.py", line 219, in _calculate_hourly_demand_forecast
    process_ok, error_txt, df_lookback = pg_select_to_pandas(cursor, sql_query, verbose=True)
  File "/opt/airflow/dags/energy.py", line 95, in pg_select_to_pandas
    cursor.connection.info.dbname + ' query: ' + sql_query +'\n' + txt
TypeError: can only concatenate str (not "NoneType") to str
[2022-10-28T22:40:26.694+0000] {taskinstance.py:1406} INFO - Marking task as UP_FOR_RETRY. dag_id=ENG_hourly_process_v1, task_id=calculate_hourly_demand_forecast, execution_date=20221026T160500, start_date=20221028T223937, end_date=20221028T224026
[2022-10-28T22:40:26.885+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 20 for task calculate_hourly_demand_forecast (can only concatenate str (not "NoneType") to str; 1161)
[2022-10-28T22:40:26.941+0000] {process_utils.py:79} INFO - Process psutil.Process(pid=1161, status='terminated', exitcode=1, started='22:39:39') (1161) terminated with exit code 1
[2022-10-28T22:40:26.945+0000] {local_task_job.py:164} INFO - Task exited with return code 143
[2022-10-28T22:40:27.163+0000] {local_task_job.py:273} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2022-10-29T00:36:02.192+0000] {taskinstance.py:1165} INFO - Dependencies all met for <TaskInstance: ENG_hourly_process_v1.calculate_hourly_demand_forecast scheduled__2022-10-26T16:05:00+00:00 [queued]>
[2022-10-29T00:36:02.299+0000] {taskinstance.py:1165} INFO - Dependencies all met for <TaskInstance: ENG_hourly_process_v1.calculate_hourly_demand_forecast scheduled__2022-10-26T16:05:00+00:00 [queued]>
[2022-10-29T00:36:02.308+0000] {taskinstance.py:1362} INFO - 
--------------------------------------------------------------------------------
[2022-10-29T00:36:02.309+0000] {taskinstance.py:1363} INFO - Starting attempt 1 of 2
[2022-10-29T00:36:02.309+0000] {taskinstance.py:1364} INFO - 
--------------------------------------------------------------------------------
[2022-10-29T00:36:02.941+0000] {taskinstance.py:1383} INFO - Executing <Task(PythonOperator): calculate_hourly_demand_forecast> on 2022-10-26 16:05:00+00:00
[2022-10-29T00:36:02.994+0000] {standard_task_runner.py:55} INFO - Started process 852 to run task
[2022-10-29T00:36:03.080+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'ENG_hourly_process_v1', 'calculate_hourly_demand_forecast', 'scheduled__2022-10-26T16:05:00+00:00', '--job-id', '37', '--raw', '--subdir', 'DAGS_FOLDER/ENG_hourly_process_dag.py', '--cfg-path', '/tmp/tmpoj3u1jzb']
[2022-10-29T00:36:03.120+0000] {standard_task_runner.py:83} INFO - Job 37: Subtask calculate_hourly_demand_forecast
[2022-10-29T00:36:03.661+0000] {task_command.py:376} INFO - Running <TaskInstance: ENG_hourly_process_v1.calculate_hourly_demand_forecast scheduled__2022-10-26T16:05:00+00:00 [running]> on host 6800cd8e39a3
[2022-10-29T00:36:04.229+0000] {taskinstance.py:1592} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=colmo786@gmail.com
AIRFLOW_CTX_DAG_OWNER=energy
AIRFLOW_CTX_DAG_ID=ENG_hourly_process_v1
AIRFLOW_CTX_TASK_ID=calculate_hourly_demand_forecast
AIRFLOW_CTX_EXECUTION_DATE=2022-10-26T16:05:00+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2022-10-26T16:05:00+00:00
[2022-10-29T00:36:04.248+0000] {logging_mixin.py:120} WARNING - /home/***/.local/lib/python3.7/site-packages/***/utils/context.py:297 AirflowContextDeprecationWarning: Accessing 'execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_start' or 'logical_date' instead.
[2022-10-29T00:36:04.248+0000] {logging_mixin.py:120} INFO - INFO calculate_hourly_demand_forecast execution init:  2022-10-26T16:05:00+00:00
[2022-10-29T00:36:04.249+0000] {logging_mixin.py:120} INFO - WORKDIR: /opt/***
[2022-10-29T00:36:04.313+0000] {logging_mixin.py:120} INFO - INFO Module build_postgres_cnxn: DB Connection to host local_pgdb Ok
[2022-10-29T00:36:04.349+0000] {logging_mixin.py:120} INFO - INFO pg_select_to_pandas: query executed Ok. Number of records returned: 1
[2022-10-29T00:36:04.391+0000] {logging_mixin.py:120} INFO - INFO pg_select_to_pandas: query executed Ok. Number of records returned: 48
[2022-10-29T00:36:27.487+0000] {logging_mixin.py:120} INFO - 1/1 [==============================] - ETA: 0s
[2022-10-29T00:36:27.488+0000] {logging_mixin.py:120} INFO - 1/1 [==============================] - 11s 11s/step
[2022-10-29T00:36:27.497+0000] {logging_mixin.py:120} INFO - INFO _calculate_hourly_demand_forecast: forecast calculated for next 24 hours. Since: 2022-10-27 00:00
[2022-10-29T00:36:35.660+0000] {logging_mixin.py:120} INFO - INFO _calculate_hourly_demand_forecast - 24 records were upserted. Table cammesa_db.hourly_demand_forecast.
[2022-10-29T00:36:35.661+0000] {python.py:177} INFO - Done. Returned value was: None
[2022-10-29T00:36:35.721+0000] {taskinstance.py:1406} INFO - Marking task as SUCCESS. dag_id=ENG_hourly_process_v1, task_id=calculate_hourly_demand_forecast, execution_date=20221026T160500, start_date=20221029T003602, end_date=20221029T003635
[2022-10-29T00:36:35.925+0000] {local_task_job.py:164} INFO - Task exited with return code 0
[2022-10-29T00:36:36.012+0000] {local_task_job.py:273} INFO - 0 downstream tasks scheduled from follow-on schedule check
